plugins/net_plugin/net_plugin.cpp

```diff
@@ -2192,13 +2192,15 @@ namespace eosio {
          controller& cc = my_impl->chain_plug->chain();
          if (cc.get_read_mode() == db_read_mode::IRREVERSIBLE) {
             auto forkdb_head = cc.fork_db_head();
-            auto calculated_lib = forkdb_head.irreversible_blocknum();
-            auto num_blocks_that_can_be_applied = calculated_lib > head_num ? calculated_lib - head_num : 0;
-            if (num_blocks_that_can_be_applied < sync_fetch_span) {
-               if (head_num )
-               fc_ilog(logger, "sync ahead allowed past sync-fetch-span ${sp}, block ${bn} for paused LIB ${l}, chain_lib ${cl}, forkdb size ${s}",
-                       ("bn", blk_num)("sp", sync_fetch_span)("l", calculated_lib)("cl", head_num)("s", cc.fork_db_size()));
-               return true;
+            if (forkdb_head.block_num() >= blk_num) {
+               auto calculated_lib = forkdb_head.irreversible_blocknum();
+               auto num_blocks_that_can_be_applied = calculated_lib > head_num ? calculated_lib - head_num : 0;
+               if (num_blocks_that_can_be_applied < sync_fetch_span) {
```

Comment:
I wonder is the test at line 2195 is not too strict. If this function is always evaluated before the last received block is added to fork_db, might we never return true?

Instead, what about the following change:
```suggestion
               auto num_blocks_that_can_be_applied = calculated_lib > head_num ? calculated_lib - head_num : 0;
               
               // add the blocks that can potentially be applied because they are not in fork_db yet
               num_blocks_that_can_be_applied += blk_num > forkdb_head.block_num() ? blk_num - forkdb_head.block_num() : 0;
               
               if (num_blocks_that_can_be_applied < sync_fetch_span) {
```

Reply:
I like that, thanks.

Reply:
Done